React/Modules/RCTTiming.m

```diff
@@ -181,22 +192,31 @@ - (void)didUpdateFrame:(__unused RCTFrameUpdate *)update
   // Call timers that need to be called
   if (timersToCall.count > 0) {
     [_bridge enqueueJSCall:@"JSTimersExecution.callTimers" args:@[timersToCall]];
-
-    // If we call at least one timer this frame, don't switch to a paused state yet, so if
-    // in response to this timer another timer is scheduled, we don't pause and unpause
-    // the displaylink frivolously.
-    return;
   }
 
-  // No need to call the pauseCallback as RCTDisplayLink will ask us about our paused
-  // status immediately after completing this call
-  if (_timers.count == 0) {
-    _paused = YES;
+  if (_sendIdleEvents) {
+    NSTimeInterval frameElapsed = (CACurrentMediaTime() - update.timestamp) * 1000;
```

Comment:
We could synchronize the clocks on each side by doing a super-fast, synchronous call from JS -> Obj-C to compute the clock skew. Once we know the clock skew, it's easy to convert Obj-C timestamps to JS timestamps. RN might also already be tracking the clock skew deep within the bridge and message queue. If so we could maybe piggyback off of that.